pkg/persistence/config/writer/config_writer.go

```diff
@@ -790,3 +790,14 @@ func fmtDetailedConfigWriterError(context *serializerContext, format string, arg
 		Err:      fmt.Errorf(format, args...),
 	}
 }
+
+var fileNameClashes = make(map[string]int)
```

Comment:
I realized that passing this through each method as a pointer "only" for identifying possible clashes pollutes the code even more. So I've decided to use that package. private global here. tell me if you think that's acceptable :)

Reply:
nit: I would go with an array of strings ( `var []string`). 
And then in the next function, use `strings.Contains....` (https://pkg.go.dev/slices#Contains)

why: there is no need to keep track of the number of duplicates.